### PR TITLE
Allow configuration of degree of parallelism of report generation for Stats.CreateAzureCdnWarehouseReports

### PR DESCRIPTION
**Pull request by scottbommarito**
We currently generate 8 reports simultaneously, but upon examination of the database, we are only using at most 10% DTU during this time (which includes reads/writes from other jobs). By increasing the amount of reports we can generate simultaneously, we should be able to drastically increase the speed of the job.

This change allows us to configure the number of reports that are generated simultaneously, so we can tune the value until we find one that gets us the maximum throughput.

Most of the changes are actually from renaming a SQL command timeout parameter as suggested by @shishirx34.

### CODE REVIEW

File: src/Stats.CreateAzureCdnWarehouseReports/Scripts/Stats.CreateAzureCdnWarehouseReports.cmd [hunk 1]
```diff
@@ -7,7 +7,7 @@ cd bin
...
+	start /w stats.createazurecdnwarehousereports.exe -VaultName "#{Deployment.Azure.KeyVault.VaultName}" -ClientId "#{Deployment.Azure.KeyVault.ClientId}" -CertificateThumbprint "#{Deployment.Azure.KeyVault.CertificateThumbprint}" -AzureCdnCloudStorageAccount "#{Jobs.stats.createazurecdnwarehousereports.AzureCdn.CloudStorageAccount}" -AzureCdnCloudStorageContainerName "#{Jobs.stats.createazurecdnwarehousereports.AzureCdn.CloudStorageContainerName}" -StatisticsDatabase "#{Jobs.stats.createazurecdnwarehousereports.StatisticsDatabase}" -SourceDatabase "#{Jobs.stats.createazurecdnwarehousereports.SourceDatabase}" -DataStorageAccount "#{Jobs.stats.createazurecdnwarehousereports.DataStorageAccount}" -InstrumentationKey "#{Jobs.stats.createazurecdnwarehousereports.InstrumentationKey}" -DataContainerName "#{Jobs.stats.createazurecdnwarehousereports.DataContainerName}" -CommandTimeOut "#{Jobs.stats.createazurecdnwarehousereports.CommandTimeOut}" -PerPackageReportDegreeOfParallelism "#{Jobs.stats.createazurecdnwarehousereports.PerPackageReportDegreeOfParallelism} -verbose true -Interval #{Jobs.stats.createazurecdnwarehousereports.Interval}
```
Comment (scottbommarito) [inline src/Stats.CreateAzureCdnWarehouseReports/Scripts/Stats.CreateAzureCdnWarehouseReports.cmd]: The only change on this line is the addition of `-PerPackageReportDegreeOfParallelism "#{Jobs.stats.createazurecdnwarehousereports.PerPackageReportDegreeOfParallelism}`

Comment (agr) [reply src/Stats.CreateAzureCdnWarehouseReports/Scripts/Stats.CreateAzureCdnWarehouseReports.cmd]: You can split the command line into separate lines and put the `^` at the end of each line (just make sure there are no spaces after the `^`, just the new line immediately). That would make next PRs bit more readable.

See:
https://github.com/NuGet/NuGet.Jobs/blob/c6bd2f6cce6d900d1c1df8f6356755eb97d39984/src/Validation.Runner/Scripts/Validation.Runner.cmd#L10-L26

Comment (joelverhagen) [reply src/Stats.CreateAzureCdnWarehouseReports/Scripts/Stats.CreateAzureCdnWarehouseReports.cmd]: Yeah, I've been doing that on all of the .cmds I touch. We'll make it there eventually.

Comment (scottbommarito) [reply src/Stats.CreateAzureCdnWarehouseReports/Scripts/Stats.CreateAzureCdnWarehouseReports.cmd]: will do

File: src/Stats.CreateAzureCdnWarehouseReports/Job.cs [hunk 2]
```diff
@@ -165,15 +168,15 @@ public override void Init(IDictionary<string, string> jobArgsDictionary)
...
-            var dirtyPackageIds = await ReportDataCollector.GetDirtyPackageIds(LoggerFactory.CreateLogger<ReportDataCollector>(), _statisticsDatabase, reportGenerationTime, _sqlCommandTimeout);
+            var dirtyPackageIds = await ReportDataCollector.GetDirtyPackageIds(LoggerFactory.CreateLogger<ReportDataCollector>(), _statisticsDatabase, reportGenerationTime, _sqlCommandTimeoutSeconds);
 
             if (!dirtyPackageIds.Any())
                 return;
 
             // first process the top 100 packages
             var top100 = dirtyPackageIds.Take(100);
-            var reportDataCollector = new ReportDataCollector(LoggerFactory.CreateLogger<ReportDataCollector>(), _storedProceduresPerPackageId[ReportNames.RecentPopularityDetailByPackageId], _statisticsDatabase, _sqlCommandTimeout);
-            var top100Task = Parallel.ForEach(top100, new ParallelOptions { MaxDegreeOfParallelism = 4 }, dirtyPackageId =>
```
Comment (shishirx34) [inline src/Stats.CreateAzureCdnWarehouseReports/Job.cs]: any reason this was 4 before? and defaulting to 8 wouldn't hamper it? I guess not just thought of pointing out.

Comment (scottbommarito) [reply src/Stats.CreateAzureCdnWarehouseReports/Job.cs]: I think the rationale was that these queries were more expensive than the others so they should be done less simultaneously, but given that DB DTU usage was so low during the execution of this as well, I didn't feel like it was important to worry.

**Pull request merged**